src/components/NewVault.tsx

```diff
@@ -10,24 +13,53 @@ const NewVault = ({ id }: Props) => {
     vaultMetrics,
     prices,
     priceErrors,
+    vaultFactoryParamsLoadingError,
   } = useVaultStore();
-
   const manager = vaultManagers.get(id);
   const metrics = vaultMetrics.get(id);
   const params = vaultGovernedParams.get(id);
   const brand = metrics?.retainedCollateral?.brand;
   const price = brand && prices.get(brand);
 
-  const error = vaultLoadingErrors.get(id) || (brand && priceErrors.get(brand));
+  const error =
+    vaultLoadingErrors.get(id) ||
+    (brand && priceErrors.get(brand)) ||
+    vaultFactoryParamsLoadingError;
+
+  const offerSigner = useAtomValue(offerSignerAtom);
+  const isReady =
+    manager &&
+    metrics &&
+    params &&
+    price &&
+    !error &&
+    offerSigner.isDappApproved;
+
+  const purses = useAtomValue(pursesAtom);
+  const istPurse = purses?.find(p => p.brand === metrics?.totalDebt?.brand);
+  const collateralPurse = purses?.find(
+    p => p.brand === metrics?.retainedCollateral?.brand,
+  );
+
+  const IST_TO_BORROW = 5_000_000n; // 5 IST
 
-  const isReady = manager && metrics && params && price && error === undefined;
+  // TODO: Calculate these based on user inputs and actual prices and liquidation ratios.
+  const proposeOffer = () => {
+    makeOpenVaultOffer(
+      collateralPurse?.pursePetname,
```

Comment:
`makeOpenVaultOffer` says the petname params are `string`. Why is it okay to pass undefined?

Reply:
I wonder why I didn't see a type error there. This is technically a bug, worst case you see faulty proposal show up in your wallet, but it would require unlucky timing for a user to get there. Fixed with an assert, this is just temporary for demo anyway.